### ISSUE TITLE
Change email error message

### ISSUE THREAD

**Issue by TimDR-1356** [supabase/supabase#19071]
When attempting to update an email to one that's already registered, an error message states, "A user with this email address has already been registered." However, when trying to sign up a user with an already registered email, no message appears, but rather a dummy user is created. Initially, I presumed this behavior was for security reasons to prevent revealing whether an account exists with a specific email. However, when attempting to change an email to a registered one, the system does provide the message indicating an existing user with that email.

I discovered this information on the sign-up process via this link: https://github.com/supabase/supabase-js/issues/296

I'm curious whether this issue is unique to my experience or if it's a widespread error that necessitates attention and resolution.

**Comment by GaryAustin1** [comment]
SignUp is the correct way Supabase wants it to work with no error.
I can't comment on intent or bug on the updateUser (I assume that is the method you used) case.

**Comment by TimDR-1356** [comment]
> SignUp is the correct way Supabase wants it to work with no error. I can't comment on intent or bug on the updateUser (I assume that is the method you used) case.

That I do understand. I'm just trying to say that the error message you get when trying to update a user's email should not be a error message but just return null where there is already a user with the email or not and not give a error saying: "A user with this email address has already been registered." Because if you leave it like this a attacker would still be able to see whether there is a user with that email.

I also think that if you leave it like this then the dummy user you get when trying to sing up with a email that already exists is useless when  the attacker could just make a account and try to change it's email, and then see if a certain user exists with a specific email.

**Comment by GaryAustin1** [comment]
I understand.  I had marked your issue with Auth.   Just was not clear to me you understood signUp was intended to work like that so just wanted to confirm that part.
I'm not a Supabase employee so can't comment on the updateUser thought or if it is an over site.

**Comment by Garett-MacGowan** [comment]
Yeah, I think this is an issue. User emails can leak. Same goes for phone auth.

@encima any progress on this behind the scenes?

**Comment by masda70** [comment]
I confirm that is still an issue ("A user with this email address has already been registered" in case of email change), and I would additionally expect that a dummy email be sent to the old address in case dual confirmations are enabled.

For the record, I noticed this behavior in the context of another issue, that I just reported:
https://github.com/supabase/supabase/issues/32909
